File: src/2/2.2/2.2-2/selection_sort.py
def selection_sort(elements):
    elementsQuantity = len(elements)
    for current in range(elementsQuantity - 1):
        smallest = current
        for selected in range(current + 1, elementsQuantity):
            if elements[selected] < elements[smallest]:
                smallest = selected
        temp_current = elements[current]
        elements[current] = elements[smallest]
        elements[smallest] = temp_current

File: src/2/2.2/2.2-2/selection-sort.ts
export function selectionSort(elements: Array<number>) {
  for (let current = 0; current < elements.length - 1; current++) {
    let smallest = current

    for (let selected = smallest + 1; selected < elements.length; selected++) {
      if (elements[selected] < elements[smallest]) {
        smallest = selected
      }
    }

    const tempCurrent = elements[current]
    elements[current] = elements[smallest]
    elements[smallest] = tempCurrent
  }

  return elements
}
